pkg/store/device/cache.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package device
+
+import (
+	"fmt"
+	networkchangestore "github.com/onosproject/onos-config/pkg/store/change/network"
+	networkchangetypes "github.com/onosproject/onos-config/pkg/types/change/network"
+	"github.com/onosproject/onos-topo/pkg/northbound/device"
+	"io"
+	"sync"
+)
+
+const separator = ":"
+
+// newKey returns a new cache key
+func newKey(id device.ID, version string) string {
+	return fmt.Sprintf("%s%s%s", id, separator, version)
+}
+
+// Info is device type/version info
+type Info struct {
+	DeviceID device.ID
+	Type     string
+	Version  string
+}
+
+// Cache is a device type/version cache
+type Cache interface {
+	io.Closer
+
+	// GetDevices returns the set of devices in the cache
+	GetDevices() []Info
+}
```

Comment:
Would be useful to also have:
GetDevices(deviceID string) - because we want to get all versions of that ID
GetDevices(deviceVersion string, deviceType string) - because we want to get all devices of that type and version
GetDevices(deviceType string) - because we want to get all devices (all versions) of that type

Reply:
okay thanks!